packages/desktop-client/src/components/mobile/MobileAmountInput.js

```diff
@@ -0,0 +1,399 @@
+    this.setState(initialState);
+
+    this.props.context.emit('reset');
+  }
+
+  applyText = () => {
+    const { editing } = this.state;
+
+    const parsed = this.parseText();
+    const newValue = editing ? parsed : getValue(this.state);
+
+    this.setState({
+      value: Math.abs(newValue),
+      editing: false,
+      text: '',
+    });
+
+    return newValue;
+  };
+
+  onBlur = () => {
+    const value = this.applyText();
+    this.props.onBlur && this.props.onBlur(value);
+    if (this.removeListeners) {
+      this.removeListeners();
+    }
+  };
+
+  onChangeText = text => {
+    let { onChange } = this.props;
+
+    this.setState({ text });
+    onChange(text);
+  };
+
+  render() {
+    const { style, textStyle, scrollIntoView } = this.props;
+    const { editing, value, text } = this.state;
+
+    let input = (
+      <input
+        type="text"
+        ref={el => (this.input = el)}
+        value={text}
+        inputMode="decimal"
+        autoCapitalize="none"
+        onChange={e => this.onChangeText(e.target.value)}
+        onBlur={this.onBlur}
+        onKeyPress={this.onKeyPress}
+        data-testid="amount-input"
+        style={{ flex: 1, textAlign: 'center', position: 'absolute' }}
+      />
+    );
+
+    return (
+      <View
+        style={[
+          {
+            justifyContent: 'center',
+            borderWidth: 1,
+            borderColor: colors.p3,
+            borderRadius: 4,
+            padding: 5,
+            backgroundColor: 'white',
+          },
+          style,
+        ]}
+      >
+        {scrollIntoView ? (
+          <View
+            style={[styles.inputContainer, styles.inputContent]}
+            pointerEvents="box-none"
+          >
+            {input}
+          </View>
+        ) : (
+          // <ScrollView
+          //   style={styles.inputContainer}
+          //   contentContainerStyle={styles.inputContent}
+          //   pointerEvents="box-none"
+          // >
+          <View style={{ overflowY: 'auto' }}>{input}</View>
+          // </ScrollView>
+        )}
+
+        {/* <Animated.View
+          style={{
+            position: 'absolute',
+            left: 0,
+            right: 0,
+            bottom: 0,
+            top: 0,
+
+            backgroundColor: animationColor || colors.p10,
+            opacity: this.backgroundValue,
+            borderRadius: 2,
+          }}
+          pointerEvents="none"
+        /> */}
+        <Text
+          style={textStyle}
+          data-testid="amount-fake-input"
+          pointerEvents="none"
+        >
+          {editing ? text : amountToCurrency(value)}
+        </Text>
+      </View>
+    );
+  }
+}
+
+const AmountInputWithContext = forwardRef((props, ref) => {
+  return (
+    <AmountAccessoryContext.Consumer>
+      {context => <AmountInput {...props} ref={ref} context={context} />}
+    </AmountAccessoryContext.Consumer>
+  );
+});
+
+class FocusableAmountInput_ extends PureComponent {
```

Comment:
Definitely more of an optional one but if you’re up to porting some or all of these to function components that would be nice! (or that can be done in a future PR/never)

Reply:
💯. I would rather do this as a follow-up; I'll add it to the list of follow-up items.